### PR TITLE
[CNF-5645] bump api version to add new capability

### PR DESCRIPTION
**Pull request by qJkee**
bump openshift/api with new capability

### CODE REVIEW

**Comment by qJkee** [comment]
Hi @rluders. I wanted to mention you because this PR is almost the same as https://github.com/openshift/cluster-version-operator/pull/817 but with newer version of `openshift/api` and one more capability

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]
Can you separate the commit for the changes around `Migrate to ConfigMapsLeasesResourceLock` . Also the version bump commit should have steps you performed e.g. 
```
$ go get github.com/openshift/api@dc2694e95008443da2f0fb2a3d2f7a552824c79d
$ go mod tidy
$ go mod vendor
$ git add -A go.* vendor
```

**Comment by qJkee** [comment]
@LalatenduMohanty I've fixed commit messages

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

File: pkg/cvo/cvo_scenarios_test.go [hunk 1]
```diff
@@ -3937,6 +3938,10 @@ func verifyAllStatusOptionalDone(t *testing.T, ignoreDone bool, ch <-chan SyncWo
 			expect.Done = actual.Done
 		}
 
+		sort.Slice(actual.CapabilitiesStatus.ImplicitlyEnabledCaps, func(i, j int) bool {
+			return actual.CapabilitiesStatus.ImplicitlyEnabledCaps[i] < actual.CapabilitiesStatus.ImplicitlyEnabledCaps[j]
```
Comment (wking) [inline pkg/cvo/cvo_scenarios_test.go:3942]: can we sort these when we set `status`, where they'd also help with stability in live clusters, and not just here in the test-case?

Comment (qJkee) [reply pkg/cvo/cvo_scenarios_test.go:3942]: sure, I'll move it

Comment (qJkee) [reply pkg/cvo/cvo_scenarios_test.go]: fixed at 54b69fcd6aba029fce086309662aa3b3c0b13386

**Comment by LalatenduMohanty** [comment]
Lets wait for merge of https://github.com/openshift/cluster-version-operator/pull/827 and then rebase this PR

**Comment by qJkee** [comment]
waiting until https://github.com/openshift/release/pull/31518 get merged to fix the upgrade test

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
openshift/release#31518 should have replaced our A->B->A presubmit with A->B and B->A presubmits.

/retest

**Comment by wking** [comment]
heh, hasn't actually landed yet 😂

**Comment by wking** [comment]
[into change][1]:

```
: Run multi-stage test pre phase
..."e2e-agnostic-upgrade-into-change" pod "e2e-agnostic-upgrade-into-change-ipi-install-install-stableinitial" failed: context canceled...
```

Not sure what that's about.  But digging in to see what's going on with versions:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-into-change/1561830992749006848/artifacts/e2e-agnostic-upgrade-into-change/gather-extra/artifacts/clusterversion.json | jq -r '.items[].status.history[] | .startedTime + " " + .completionTime + " " + .state + " " + .version'
2022-08-22T22:06:47Z 2022-08-22T22:42:12Z Completed 4.12.0-0.ci.test-2022-08-22-214209-ci-op-3mwr1wss-initial
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-into-change/1561830992749006848/artifacts/e2e-agnostic-upgrade-into-change/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-74b4d84766-pkscn_cluster-version-operator.log | head -n1
I0822 22:27:38.568033       1 start.go:23] ClusterVersionOperator v1.0.0-878-g869df9e1-dirty
$ git --no-pager log --oneline -1 origin/master
869df9e1 (HEAD -> master, origin/release-4.13, origin/release-4.12, origin/master, origin/HEAD) Merge pull request #827 from wking/more-cap-sorting
```

So hooray, it was trying to install the branch-tip A, not the release B with this change mixed in.

[out of change][2]:

```
: Run multi-stage test test phase expand_less | 33m55s
-- | --
{  ["e2e-agnostic-upgrade-out-of-change" pod "e2e-agnostic-upgrade-out-of-change-openshift-e2e-test" failed: context canceled
...
```

Maybe the test-platform folks had some kind of node issue that caused them to evict running tests?  Anyway, digging in to versions:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-out-of-change/1561830992786755584/artifacts/e2e-agnostic-upgrade-out-of-change/gather-extra/artifacts/clusterversion.json | jq -r '.items[].status.history[] | .startedTime + " " + .completionTime + " " + .state + " " + .version'
2022-08-22T22:30:54Z  Partial 4.12.0-0.ci.test-2022-08-22-215104-ci-op-3mwr1wss-latest
2022-08-22T21:59:50Z 2022-08-22T22:23:58Z Completed 4.12.0-0.ci.test-2022-08-22-214209-ci-op-3mwr1wss-initial
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-out-of-change/1561830992786755584/artifacts/e2e-agnostic-upgrade-out-of-change/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-66d4c75496-fhdqg_cluster-version-operator.log | head -n1
I0822 22:30:56.387655       1 start.go:23] ClusterVersionOperator v1.0.0-882-gb44878b1-dirty
```

Hmm, looks like that might be doing an A->B update too.  I'll go back and look at openshift/release#31518 to see if I need to change something.  In the mean time:

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-into-change/1561830992749006848
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/801/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade-out-of-change/1561830992786755584

**Comment by wking** [comment]
C'mon Prow, it doesn't exist anymore...

image

Just in case:

/test ci/prow/e2e-agnostic-upgrade

**Pull request merged**